tests/interpreters/test_onnx_interpreter.py

```diff
@@ -0,0 +1,80 @@
+from xdsl.dialects import onnx
+from xdsl.dialects.builtin import ModuleOp, TensorType, f32
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.onnx import OnnxFunctions
+from xdsl.interpreters.shaped_array import ShapedArray
+from xdsl.utils.test_value import TestSSAValue
+
+interpreter = Interpreter(ModuleOp([]))
+interpreter.register_implementations(OnnxFunctions())
+
+
+def test_onnx_add():
+    op = onnx.Add(
+        TestSSAValue(TensorType(f32, [2, 3])),
+        TestSSAValue(TensorType(f32, [2, 3])),
+        res_type=TensorType(f32, [2, 3]),
+    )
+
+    a = ShapedArray([1, 2, 3, 4, 5, 6], [2, 3])
+    b = ShapedArray([1, 4, 2, 5, 3, 6], [2, 3])
+    c = ShapedArray([2, 6, 5, 9, 8, 12], [2, 3])
+
+    interpreter.run_op(op, (a, b, c))
+    assert c.data == [2, 6, 5, 9, 8, 12]
+
+
+def test_onnx_sub():
+    op = onnx.Sub(
+        TestSSAValue(TensorType(f32, [2, 3])),
+        TestSSAValue(TensorType(f32, [2, 3])),
+        res_type=TensorType(f32, [2, 3]),
+    )
+
+    a = ShapedArray([1, 2, 3, 4, 5, 6], [2, 3])
+    b = ShapedArray([1, 4, 2, 5, 3, 6], [2, 3])
+    c = ShapedArray([0, -2, 1, -1, 2, 0], [2, 3])
```

Comment:
```suggestion
    c = ShapedArray([0, 0, 0, 0, 0, 0], [2, 3])
```

Reply:
We want to test that we get the expected result in the end

Reply:
but isn't that the expected result, if you subtract a and b? i don't get why [0,0,0,0..]

Reply:
because you're giving the result as the input, in real life the input will either be initialized to random or 0, and has to be set as part of the calculation